src/main/java/seedu/address/ui/MainWindowNew.java

```diff
@@ -0,0 +1,220 @@
+
+    /**
+     * Creates a {@code MainWindow} with the given {@code Stage} and {@code Logic}.
+     */
+    public MainWindowNew(Stage primaryStage, Logic logic) {
+        super(FXML, primaryStage);
+
+        // Set dependencies
+        this.primaryStage = primaryStage;
+        this.logic = logic;
+
+        // Configure the UI
+        setWindowDefaultSize(logic.getGuiSettings());
+
+        setAccelerators();
+        setAcceleratorsBugReport();
+
+        helpWindow = new HelpWindow();
+        reportBugWindow = new ReportBugWindow();
+
+    }
+
+    public Stage getPrimaryStage() {
+        return primaryStage;
+    }
+
+    private void setAccelerators() {
+        setAccelerator(helpMenuItem, KeyCombination.valueOf("F1"));
+    }
+
+    private void setAcceleratorsBugReport() {
+        setAccelerator(reportBugItem, KeyCombination.valueOf("F2"));
+    }
+
+    /**
+     * Sets the accelerator of a MenuItem.
+     * @param keyCombination the KeyCombination value of the accelerator
+     */
+    private void setAccelerator(MenuItem menuItem, KeyCombination keyCombination) {
+        menuItem.setAccelerator(keyCombination);
+
+        /*
+         * TODO: the code below can be removed once the bug reported here
+         * https://bugs.openjdk.java.net/browse/JDK-8131666
+         * is fixed in later version of SDK.
+         *
+         * According to the bug report, TextInputControl (TextField, TextArea) will
+         * consume function-key events. Because CommandBox contains a TextField, and
+         * ResultDisplay contains a TextArea, thus some accelerators (e.g F1) will
+         * not work when the focus is in them because the key event is consumed by
+         * the TextInputControl(s).
+         *
+         * For now, we add following event filter to capture such key events and open
+         * help window purposely so to support accelerators even when focus is
+         * in CommandBox or ResultDisplay.
+         */
+        getRoot().addEventFilter(KeyEvent.KEY_PRESSED, event -> {
+            if (event.getTarget() instanceof TextInputControl && keyCombination.match(event)) {
+                menuItem.getOnAction().handle(new ActionEvent());
+                event.consume();
+            }
+        });
+    }
+
+    void fillInnerParts() {
+
+        this.resultDisplay = new ResultDisplay();
+        resultDisplayPlaceholder.getChildren().add(resultDisplay.getRoot());
+
+        this.commandBox = new CommandBox(this::executeCommand);
+        commandBoxPlaceholder.getChildren().add(commandBox.getRoot());
+
+        this.overviewPanel = new OverviewPanel();
+        overviewPanelPlaceholder.getChildren().add(overviewPanel.getRoot());
+
+        this.personListPanel = new PersonListPanel(logic.getFilteredPersonList(), overviewPanel);
+        personListPanelPlaceholder.getChildren().add(personListPanel.getRoot());
+
+    }
+
+    private void setWindowDefaultSize(GuiSettings guiSettings) {
```

Comment:
Method adheres to SLAP to a reasonable extent - could it be abstracted even further?

Reply:
Does the method adhere to the Law of Demeter?